service/matching/task_queue_manager.go

```diff
@@ -146,6 +147,8 @@ type (
 		QueueID() *taskQueueID
 		TaskQueueKind() enumspb.TaskQueueKind
 		LongPollExpirationInterval() time.Duration
+		RedirectToVersionedQueueForAdd(context.Context, *taskqueuespb.TaskVersionDirective) (*taskQueueID, chan struct{}, error)
+		RedirectToVersionedQueueForPoll(*commonpb.WorkerVersionCapabilities) (*taskQueueID, error)
```

Comment:
Not related to this PR, but it seems like we can get rid of this interface? There's only one implementation.

Reply:
Probably! Feel free to take a crack at it, or I might in a while. I think it could theoretically help with tests reaching into internals, but... they do it anyway with type assertions :(